packages/elements/src/datetime-picker/index.ts

```diff
@@ -1088,6 +1093,34 @@ export class DatetimePicker extends FormFieldElement implements MultiValue {
     this.validateInput();
   }
 
+  /**
+   * Run when tap event happens on clears button and fire value-changed event.
+   * @param event tap event
+   * @returns {void}
+   */
+  protected override onClearsButtonTap(event: TapEvent): void {
+    super.onClearsButtonTap(event);
+    this.inputValues = [];
+    if (this.inputFromEl && this.inputToEl) {
+      this.inputFromEl.value = '';
+      this.inputToEl.value = '';
+    } else if (this.inputEl) {
+      this.inputEl.value = '';
+    }
+    this.syncInputValues();
+    this.resetViews();
+  }
+
+  /**
+   * Returns a condition to show clear button
+   * @returns True if clear button will show
+   */
+  protected override get hasClear(): boolean {
+    const editable = !(this.readonly || this.disabled); // shouldn't display clear if disabled or readonly
+    const hasInputValue = this.range ? !!this.inputValues[0] || !!this.inputValues[1] : !!this.inputValues[0];
+    return this.clears && editable && this.inputValues.length > 0 && hasInputValue;
```

Comment:
`inputValues.length` check is necessary since `hasInputValue` could do just that. `[][1]` would evaluates to `undefined`.

Reply:
From this
`return this.clears && editable && this.inputValues.length > 0 && hasInputValue;`
to this
`return this.clears && editable && hasInputValue;`
 should be enough already.

As `values` is an API, it might be set to an array with single element e.g. `['2014-01-05']`. It's better to support such use case if we could. Not to mention that the logic is simpler.